### PR TITLE
Converts `aws_appconfig_environment` to Framework and AWS SDK for Go v2

### PR DESCRIPTION
**Pull request by gdavison**
### Description

Converts `aws_appconfig_environment` to Framework and AWS SDK for Go v2.

### Output from Acceptance Testing

```
$ make testacc PKG=appconfig TESTS=TestAccAppConfigEnvironment 

--- PASS: TestAccAppConfigEnvironmentsDataSource_basic (30.83s)

--- PASS: TestAccAppConfigEnvironmentDataSource_basic (31.98s)

--- PASS: TestAccAppConfigEnvironment_disappears (30.43s)
--- PASS: TestAccAppConfigEnvironment_basic (33.79s)
--- PASS: TestAccAppConfigEnvironment_updateName (45.55s)
--- PASS: TestAccAppConfigEnvironment_multipleEnvironments (54.12s)
--- PASS: TestAccAppConfigEnvironment_tags (60.27s)
--- PASS: TestAccAppConfigEnvironment_updateDescription (62.10s)
--- PASS: TestAccAppConfigEnvironment_monitors (64.70s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Want to do the AWS SDK for Go v2 migration at the same time 😸?

File: internal/service/appconfig/environment.go [hunk 1]
```diff
@@ -3,333 +3,476 @@ package appconfig
...
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"arn": {
-				Type:     schema.TypeString,
+			"description": schema.StringAttribute{
+				Optional: true,
 				Computed: true,
+				Default:  stringdefault.StaticString(""),
```
Comment (jar-b) [inline internal/service/appconfig/environment.go:76]: Is this for Plugin-SDK backward compatibility?

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

Would it be worth adding a "migration" acceptance test which creates the resource with an older, Plugin-SDK version, then makes an update with this?

```console
$ make testacc PKG=appconfig TESTS=TestAccAppConfigEnvironment
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/appconfig/... -v -count 1 -parallel 20 -run='TestAccAppConfigEnvironment'  -timeout 180m

--- PASS: TestAccAppConfigEnvironment_disappears (22.86s)
--- PASS: TestAccAppConfigEnvironmentsDataSource_basic (24.01s)
--- PASS: TestAccAppConfigEnvironmentDataSource_basic (24.57s)
--- PASS: TestAccAppConfigEnvironment_basic (27.07s)
--- PASS: TestAccAppConfigEnvironment_updateName (39.08s)
--- PASS: TestAccAppConfigEnvironment_multipleEnvironments (45.47s)
--- PASS: TestAccAppConfigEnvironment_tags (51.73s)
--- PASS: TestAccAppConfigEnvironment_updateDescription (54.31s)
--- PASS: TestAccAppConfigEnvironment_monitors (56.12s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/appconfig  59.592s
```

File: internal/service/appconfig/environment.go (hunk 1 above)
Comment (gdavison) [reply internal/service/appconfig/environment.go:76]: Yes. I'll add a comment

**Comment by gdavison** [comment]
> Would it be worth adding a "migration" acceptance test which creates the resource with an older, Plugin-SDK version, then makes an update with this?

Yes, good catch

**Comment by gdavison** [comment]
> Want to do the AWS SDK for Go v2 migration at the same time 😸?

I was planning to do a follow-up PR for that to isolate the changes and sources of potential errors

**Comment by gdavison** [comment]
```
--- PASS: TestAccAppConfigEnvironment_frameworkMigration_basic (45.53s)
--- PASS: TestAccAppConfigEnvironment_frameworkMigration_monitors (46.84s)
```

**Review by jar-b — APPROVED** [review]
LGTM 🚀 

```console
make testacc PKG=appconfig TESTS=TestAccAppConfigEnvironment
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/appconfig/... -v -count 1 -parallel 20 -run='TestAccAppConfigEnvironment'  -timeout 180m

--- PASS: TestAccAppConfigEnvironment_disappears (25.96s)
--- PASS: TestAccAppConfigEnvironmentsDataSource_basic (26.55s)
--- PASS: TestAccAppConfigEnvironmentDataSource_basic (27.74s)
--- PASS: TestAccAppConfigEnvironment_basic (29.29s)
--- PASS: TestAccAppConfigEnvironment_updateName (43.29s)
--- PASS: TestAccAppConfigEnvironment_frameworkMigration_monitors (46.16s)
--- PASS: TestAccAppConfigEnvironment_frameworkMigration_basic (46.62s)
--- PASS: TestAccAppConfigEnvironment_multipleEnvironments (49.44s)
--- PASS: TestAccAppConfigEnvironment_tags (56.94s)
--- PASS: TestAccAppConfigEnvironment_updateDescription (59.92s)
--- PASS: TestAccAppConfigEnvironment_monitors (60.95s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/appconfig  64.297s
```

**Comment by gdavison** [comment]
Actually, the AWS SDK for Go v2 port is really easy, so I'm including it

**Pull request merged**